CMakeLists.txt

```diff
@@ -141,6 +141,8 @@ else()
   include_directories(SYSTEM ${MLIR_TABLEGEN_OUTPUT_DIR})
 
   set(BACKEND_PACKAGE_STRING "${PACKAGE_STRING}")
+
+  set(CIRCT_GTEST_AVAILABLE 1)
```

Comment:
If I'm reading the `if` scopes properly, this seems to be assuming gtest is available when it may not be.

Reply:
Hmm, if it's possible we could have a check (thanks!) but I'm not sure that's needed here for same reason LLVM itself doesn't check for its bundled gtest bits AFAIK.
This branch should be in the "LLVM is the root project" situation.
Short of building from an incomplete source tree (not unheard of when the monorepo is too big, but not sure can omit the gtest bits) is there a way LLVM's bundled gtest might not be available? :thinking:

Reply:
Unfortunately, that's not the case :disappointed: but sure seems like how it ideally would work.
IIRC not only is it [vendored](https://github.com/llvm/llvm-project/tree/1c41d0cb62c2637b90a3ae954603af7ba9984975/llvm/utils/unittest/googletest), if you try to use an external version you'll have a bad time since the included version has changes needed for tests to work.

A pain point is none of that is installed, so it becomes a bit awkward for downstream projects wanting to use it for testing w/o the source tree.  When building separately I actually grab just that directory and point projects at it so they can find what they need for tests without the full tree :upside_down_face: .